be/src/olap/tablet.cpp

```diff
@@ -826,6 +832,33 @@ void Tablet::_print_missed_versions(const std::vector<Version>& missed_versions)
     LOG(WARNING) << ss.str();
 }
 
+bool Tablet::_drop_rowset_if_version_conflict(const RowsetSharedPtr& rowset) {
```

Comment:
There has a problem in this place.
modify_rowsets() only change the TabletMeta in memory. But not change the meta in disk.
So you should perform do_tablet_meta_checkpoint to save the meta and remove the rowsetmeta in RowsetMetaManager.

Reply:
Why don't we need to do `do_tablet_meta_checkpoint` when adding a rs, but we need to do it when we delete a rs?

Reply:
Because before add the rowset, it has already stills in RowsetMetaManager.
It only make it visible in memory.
After that, if you want to delete the rowset, you should delete from the RowsetMetaManager and add it to TabletMeta and save it.